src/snowflake/cli/plugins/nativeapp/codegen/sandbox.py

```diff
@@ -0,0 +1,201 @@
+import os
+import shutil
+import subprocess
+import sys
+from enum import Enum
+from pathlib import Path
+from typing import Optional, Sequence, Union
+
+from click.exceptions import ClickException
+
+
+class SandboxExecutionError(ClickException):
+    """An error occurred while executing a python script."""
+
+    def __init__(self, error: str):
+        super().__init__(f"Failed to execute python script. {error}")
+
+
+def _get_active_venv_dir() -> Optional[str]:
+    return os.environ.get("VIRTUAL_ENV")
+
+
+def _get_active_conda_env() -> Optional[str]:
+    return os.environ.get("CONDA_DEFAULT_ENV")
+
+
+def _is_venv_active() -> bool:
+    return _get_active_venv_dir() is not None
+
+
+def _is_conda_active() -> bool:
+    return _get_active_conda_env() is not None
+
+
+def _is_ms_windows() -> bool:
+    return sys.platform == "win32"
+
+
+def _execute_python_interpreter(
+    python_executable: Optional[Union[str, Path, Sequence[str]]],
+    script_source: str,
+    cwd: Optional[Union[str, Path]],
+    timeout: Optional[int],
+) -> subprocess.CompletedProcess:
+    if not python_executable:
+        raise SandboxExecutionError("No python executable found")
+
+    if isinstance(python_executable, str) or isinstance(python_executable, Path):
+        args = [python_executable]
+    else:
+        args = [arg for arg in python_executable]
+    args.append("-")
+    return subprocess.run(
+        args,
+        capture_output=True,
+        text=True,
+        input=script_source,
+        timeout=timeout,
+        cwd=cwd,
+    )
+
+
+def _execute_in_venv(
+    script_source: str,
+    venv_path: Optional[Union[str, Path]] = None,
```

Comment:
`venv_path` is determined from the PDF, and passed on to this function as a string. Is using `Path` as a type needed here?

Reply:
I wrote the utility function to be generally useful. Felt like a minor decision that can increase reusability.

Reply:
Ack